### DISCUSSION TITLE
how to enable GPU acceleration on ARM

### DISCUSSION THREAD

**Discussion by ebetzler** [AvaloniaUI/Avalonia#10839] [General]
Device: iMX6
OS: arm-linux
I have a .NET7-avaloniaUI application that is rendering slow when I publish/deploy to my ARM processor (iMX6).
I have done quite a bit of searching and cannot find how to enable OpenGL rendering for my ARM build.
Per this discussion:
[https://github.com/AvaloniaUI/Avalonia/issues/5375](url)
I have done this:
```
            var t = typeof(Avalonia.OpenGL.IPlatformOpenGlInterface);
            var enabled = (Avalonia.AvaloniaLocator.Current.GetService(t) != null);
           // on linux desktop pc: result is enabled=true
           // on arm iMX6: result is enabled=false
```
Can anyone tell me how to accomplish GPU/OpenGL rendering on my linux-arm build?

**Comment by kekekeks** [comment]
Show your `glxinfo`

**Reply by ebetzler** [reply]
Does it matter that we are using XWayland? I've been picking up bits and pieces that make it sound like Avalonia does not support GPU acceleration on XWayland.  If not.. are there other alternatives?

**Reply by kekekeks** [reply]
Avalonia uses GLX. If your system provides support for GLX, Avalonia will use it. Please, show your `glxinfo`
